Enable horizontal legends

jtroberts opened the pull request:
## Purpose
- Enable an option to display legends horizontally in addition to the default of vertically.
## Proposed Changes
- New configuration option to select horizontal or vertical legend
- Display continuous or discrete legends with ticks and appropriate label sizes (will condense labels if too many for display)
- Condense units and display at the top of the legend to avoid label cluttering
- Show tooltips of legend values when hovering over the legends (any type of legend including vertical)
## Issues
- (https://github.com/NASA-AMMOS/MMGIS/issues/756)
## Testing
- Tested with various legend types and ensured existing legends work properly.
- Examples:

<img width="333" height="198" alt="Screenshot 2025-09-22 at 7 03 54 PM" src="https://github.com/user-attachments/assets/16a11344-c3a1-4759-b1ea-431a82fbaf3b" />
<p>
<img width="336" height="160" alt="Screenshot 2025-09-22 at 6 57 26 PM" src="https://github.com/user-attachments/assets/abb3e438-78c4-4c25-882b-ea786573768d" />

jtroberts committed:
#756 Enable horizontal legends

jtroberts committed:
#756 Improvements to horizontal legend rendering and added tooltips

tariqksoliman reviewed (changes requested):
LGTM!

Only two things of note I found:
- When the legend orientation is horizontal, the definitions for continuous legends seem inverted than what I'd expect. 0 -> 100 (top -> bottom) in the legend configuration but 100 -> 0 in the legend tool. 
- The hover over text seems to be relying entirely on the legends "Label" configuration for continuous legends. I think it should prioritize "Property Value" else "Label".

tariqksoliman commented:
Maybe the first bullet point isn't a problem -- it does make sense to have the higher values on top.

jtroberts committed:
#756 Prioritize propertyValue on hover and fixed missing units label for non-continuous

jtroberts commented:
> Maybe the first bullet point isn't a problem -- it does make sense to have the higher values on top.

That makes more sense to me so I left it as-is. 

> * The hover over text seems to be relying entirely on the legends "Label" configuration for continuous legends. I think it should prioritize "Property Value" else "Label".

I've updated the hover behavior to show the "raw" property value.

Also fixed an issue I found where the units would display only for continuous legends.

tariqksoliman reviewed (approved):
Thank you and LGTM now!
